store/sqlstore/role_store.go

```diff
@@ -149,27 +148,24 @@ func (s *SqlRoleStore) createRole(role *model.Role, transaction *gorp.Transactio
 	return dbRole.ToModel(), nil
 }
 
-func (s *SqlRoleStore) Get(roleId string) (*model.Role, *model.AppError) {
+func (s *SqlRoleStore) Get(roleId string) (*model.Role, error) {
 	var dbRole Role
 
 	if err := s.GetReplica().SelectOne(&dbRole, "SELECT * from Roles WHERE Id = :Id", map[string]interface{}{"Id": roleId}); err != nil {
 		if err == sql.ErrNoRows {
-			return nil, model.NewAppError("SqlRoleStore.Get", "store.sql_role.get.app_error", nil, "Id="+roleId+", "+err.Error(), http.StatusNotFound)
+			return nil, store.NewErrNotFound("Role", roleId)
 		}
-		return nil, model.NewAppError("SqlRoleStore.Get", "store.sql_role.get.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return nil, errors.Wrap(err, "failed to get Role")
 	}
 
 	return dbRole.ToModel(), nil
 }
 
-func (s *SqlRoleStore) GetAll() ([]*model.Role, *model.AppError) {
+func (s *SqlRoleStore) GetAll() ([]*model.Role, error) {
 	var dbRoles []Role
 
 	if _, err := s.GetReplica().Select(&dbRoles, "SELECT * from Roles", map[string]interface{}{}); err != nil {
-		if err == sql.ErrNoRows {
-			return nil, model.NewAppError("SqlRoleStore.GetAll", "store.sql_role.get_all.app_error", nil, err.Error(), http.StatusNotFound)
-		}
-		return nil, model.NewAppError("SqlRoleStore.GetAll", "store.sql_role.get_all.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return nil, errors.Wrap(err, "failed to find Roles")
```

Comment:
Aren't we supposed to return a store.ErrNotFound here?

Reply:
Afaik Select doesn't return ErrNoRows

Reply:
Ah that's right. Thanks.